invenio_app_ils/ui/backoffice/src/pages/UserDetails/components/PatronLoansTable/PatronLoansTable.js

```diff
@@ -0,0 +1,245 @@
+  };
+
+  _getFormattedDate = d => (d ? toString(d) : '');
+
+  _renderLoan = loan => {
+    const updated = this._getFormattedDate(loan.updated);
+    const startDate = this._getFormattedDate(loan.start_date);
+    const endDate = this._getFormattedDate(loan.end_date);
+    return (
+      <Table.Row key={loan.loan_pid} data-test={loan.loan_pid}>
+        <Table.Cell textAlign="center" collapsing>
+          <Icon
+            circular
+            name="eye"
+            link
+            onClick={() => {
+              this._handleRowClick(loan.loan_pid);
+            }}
+          />
+        </Table.Cell>
+        <Table.Cell textAlign="center" collapsing>
+          {loan.loan_pid}
+        </Table.Cell>
+        <Table.Cell textAlign="center" collapsing>
+          {loan.item_pid}
+        </Table.Cell>
+        <Table.Cell textAlign="center" collapsing>
+          {loan.state}
+        </Table.Cell>
+        <Table.Cell textAlign="center" collapsing>
+          {updated}
+        </Table.Cell>
+        <Table.Cell>{startDate}</Table.Cell>
+        <Table.Cell>{endDate}</Table.Cell>
+      </Table.Row>
+    );
+  };
+
+  _handleShowAllClick = () => {
+    const { document_pid, item_pid } = this.props.item
+      ? this.props.item.metadata
+      : this.initProps;
+    const patron_pid = this.props.patron ? this.props.patron : null;
+    const loan_state = this.props.loanState ? this.props.loanState : null;
+    const qs = loanApi.buildLoansQuery(
+      document_pid,
+      item_pid,
+      loan_state,
+      patron_pid
+    );
+    const url = `${URLS.loansSearch}?q=${qs}`;
+    this.props.history.push(url);
+  };
+
+  _renderFooter = loans => {
+    const total = loans.length;
+    return total > this.props.showMaxLoans ? (
+      <Table.Footer fullWidth className="footer" data-test="footer">
+        <Table.Row>
+          <Table.HeaderCell colSpan="6" />
+          <Table.HeaderCell textAlign="right">
+            <span>Showing first {this.props.showMaxLoans} user requests</span>
+            <Button size="small" onClick={this._handleShowAllClick}>
+              Show all
+            </Button>
+          </Table.HeaderCell>
+        </Table.Row>
+      </Table.Footer>
+    ) : null;
+  };
+
+  _handleSortClick = sortByValue => {
+    const { document_pid, item_pid } = this.props.item
+      ? this.props.item.metadata
+      : this.initProps;
+    const patron_pid = this.props.patron ? this.props.patron : null;
+    const loan_state = this.props.loanState ? this.props.loanState : null;
+    if (this.props.currentSortBy === sortByValue) {
+      // change sort order
+      this.patronLoansChangeSortOrder(
+        document_pid,
+        item_pid,
+        loan_state,
+        patron_pid
+      );
+    } else {
+      // change sort by
+      this.patronLoansChangeSortBy(
+        document_pid,
+        item_pid,
+        loan_state,
+        patron_pid
+      );
+    }
+  };
+
+  _renderLoans = loans => {
```

Comment:
I think that the reusable stuff should be only the way that the results will be presented...So this component just expects some props in order to render the results...I will agree with @ntarocco on this one that redux should come from the component that will use this one along with the props needed.